### PR TITLE
Embedded cluster working group

### PR DESCRIPTION
**Pull request by laverya**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/embeddedcluster/helmvm_node.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	if err != nil {
+		fmt.Printf("failed to parse role label %q: %s", numRolesStr, err.Error())
+
+		return nil
+	}
+
+	for i := 0; i < numRoles; i++ {
+		roleLabel, ok := labels[fmt.Sprintf("%s-%d", types.EMBEDDED_CLUSTER_ROLE_LABEL, i)]
+		if !ok {
+			fmt.Printf("failed to find role label %d", i)
```
Comment (sgalsaleh) [inline pkg/embeddedcluster/helmvm_node.go:162]: should there be a `continue` after this?

File: pkg/embeddedcluster/node_join.go [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+								},
+							},
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:    "k0s-token-generator",
+					Image:   "ubuntu:latest", // TODO use the kotsadm image here as we'll know it exists
```
Comment (sgalsaleh) [inline pkg/embeddedcluster/node_join.go:146]: is this a "we'll do this in a separate" pr type of TODO?

File: pkg/handlers/embedded_cluster_node_join_command.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+func (h *Handler) GenerateEmbeddedClusterNodeJoinCommand(w http.ResponseWriter, r *http.Request) {
+	generateEmbeddedClusterNodeJoinCommandRequest := GenerateEmbeddedClusterNodeJoinCommandRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&generateEmbeddedClusterNodeJoinCommandRequest); err != nil {
+		logger.Error(fmt.Errorf("failed to decode request body: %w", err))
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	store := kotsstore.StoreFromEnv()
+	token, err := store.SetEmbeddedClusterInstallCommandRoles(generateEmbeddedClusterNodeJoinCommandRequest.Roles)
```
Comment (sgalsaleh) [inline pkg/handlers/embedded_cluster_node_join_command.go:37]: And I'd delete the line above
```suggestion
	token, err := store.GetStore().SetEmbeddedClusterInstallCommandRoles(generateEmbeddedClusterNodeJoinCommandRequest.Roles)
```

File: web/src/components/apps/EmbeddedClusterViewNode.jsx [hunk 4]
```diff
@@ -0,0 +1,245 @@
+import { MaterialReactTable } from "material-react-table";
+import React, { useMemo } from "react";
+import { useQuery } from "@tanstack/react-query";
+import { Link, useParams } from "react-router-dom";
+import Loader from "@components/shared/Loader";
+
+const testData = undefined;
+// const testData = {
```
Comment (sgalsaleh) [inline web/src/components/apps/EmbeddedClusterViewNode.jsx:8]: ditto

File: pkg/embeddedcluster/node_join.go (hunk 2 above)
Comment (laverya) [reply pkg/embeddedcluster/node_join.go:146]: haaaaaah
no that's a "we intended to be using a different method that didn't happen yet" type of TODO
let me see how much effort that'll take

File: pkg/embeddedcluster/helmvm_node.go (hunk 1 above)
Comment (laverya) [reply pkg/embeddedcluster/helmvm_node.go:162]: yes, good catch

**Review by sgalsaleh — DISMISSED** [review]

File: .github/workflows/regression.yaml [hunk 5]
```diff
@@ -191,9 +191,9 @@ jobs:
             is_upgrade: "1"
           },
           {
-            name: "type=helmvm cluster, env=online, phase=new install, rbac=cluster admin",
-            backend_config: "helmvm-online-install-backend-config.tfvars",
-            terraform_script: "helmvm-online-install.sh"
+            name: "type=embeddedcluster cluster, env=online, phase=new install, rbac=cluster admin",
```
Comment (sgalsaleh) [inline .github/workflows/regression.yaml:194]: hmm, i don't think there's such a test in https://github.com/replicatedhq/kots-regression-automation or in testim

Comment (laverya) [reply .github/workflows/regression.yaml:194]: I'll undo it

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**